src/libraries/System.Text.Encoding.CodePages/src/System/Text/DBCSCodePageEncoding.cs

```diff
@@ -135,7 +135,8 @@ protected override unsafe void LoadManagedCodePage()
                 lock (s_streamLock)
                 {
                     s_codePagesEncodingDataStream.Seek(m_firstDataWordOffset, SeekOrigin.Begin);
-                    s_codePagesEncodingDataStream.Read(buffer, 0, m_dataSize);
+                    int bytesRead = s_codePagesEncodingDataStream.Read(buffer, 0, m_dataSize);
+                    Debug.Assert(bytesRead == m_dataSize);
```

Comment:
Consider adding a message to these asserts. Someone might come along later and remove them as erroneous without more explanation.